### PR TITLE
chore: README improvements

### PR DESCRIPTION
**Pull request by chrishutchinson**
**What**  
This PR includes a number of changes:
* Improved clarity around dependencies and installation steps
* Separation of application-specific env vars from Cypress-specific env vars
* Documentation in the `README` for each environment variable, and what it's used for
* Added some specifics about what this application's responsibilities are
* Removes the `package-lock.json` file that had errantly made it into the `main` branch (we use `yarn`, not `npm`!)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,105 @@
...
+| AUTHORISED_ALLOCATORS_GROUP     | The name of the Allocators Google Group                                                       |
+| AUTHORISED_UNRESTRICTED_GROUP   | The name of the Unrestricted Google Group                                                     |
+| AUTHORISED_DEV_GROUP            | The name of the Developers Google Group                                                       |
+| POSTCODE_LOOKUP_URL             | The URL for the postcode lookup service (including `/api/v1/addresses?postcode=` at the end!) |
+| POSTCODE_LOOKUP_APIKEY          | The API key for the postcode lookup service (defined in `POSTCODE_LOOKUP_URL`)                |
+| REDIRECT_URL                    | The URL to redirect to post-authentication (`dev.hackney.gov.uk:3000` for local development)  |
+| NEXT_PUBLIC_GOOGLE_ANALYTICS_ID | The ID for Google Analytics data to be reported to                                            |
+| NEXT_PUBLIC_FEEDBACK_LINK       | The Google Form URL that powers the feedback link                                             |
+| STORYBOOK_API_PROXY             | ???                                                                                           |
```
Comment (chrishutchinson) [inline README.md:49]: Does anyone know if this is actually used? I'll remove if not

Comment (neilkidd) [reply README.md:49]: What do you think of moving the descriptions / comments into the `env.sample` file?
We can still point to the file , the comments remain close to the usage and would be easier to maintain

**Review by neilkidd — APPROVED** [review]
One non blocking suggestion. Otherwise :shipit: 💯

File: README.md (hunk 1 above)
Comment (chrishutchinson) [reply README.md]: Yeah really nice idea @neilkidd – done in https://github.com/LBHackney-IT/lbh-social-care-frontend/pull/454/commits/bd94500ddb8d2f1492fd98c083905989377c291d. Let me know what you think?

Comment (neilkidd) [reply README.md]: 🏆 🥇

**Pull request merged**